### ISSUE TITLE
StreamData spoofing flag is set too aggressively with multiple source or multiple forecast steps

### ISSUE THREAD

**Issue by clessig** [ecmwf/WeatherGenerator#1117]
### What happened?

Currently, the spoofing flag is set disregarding the [loop over potentially multiple source files](https://github.com/ecmwf/WeatherGenerator/blob/638ca1603afd79d26b12a7a8333ca52e1c130143/src/weathergen/datasets/multi_stream_data_sampler.py#L339) (as used with SEVIRI) and [over multiple forecast steps](https://github.com/ecmwf/WeatherGenerator/blob/638ca1603afd79d26b12a7a8333ca52e1c130143/src/weathergen/datasets/multi_stream_data_sampler.py#L373). This leads to incorrect behaviour when:
- There are multiple source files but the last one does not overlap the current sample window (in which case [this](https://github.com/ecmwf/WeatherGenerator/blob/638ca1603afd79d26b12a7a8333ca52e1c130143/src/weathergen/datasets/multi_stream_data_sampler.py#L368) is set)
- Training has multiple forecast steps and the last one is empty 

Also, there should be no dependence of `stream_data.is_spoof` on source since the [problem with FSDP2](https://github.com/pytorch/pytorch/issues/158719) requires gradients to be present, but no source is needed for this (as is the case when a stream is trained diagnostically).
